openwisp_controller/config/admin.py

```diff
@@ -859,29 +859,83 @@ def create_log_entry(user, clone):
                 action_flag=ADDITION,
             )
 
+        def save_clones(view, user, queryset, organization=None):
+            errors = []
+            # validate organization
+            if organization:
+                try:
+                    validated_org = Organization.objects.get(pk=organization)
+                except (ValidationError, Organization.DoesNotExist) as e:
+                    logger.warn(
+                        'Detected tampering in clone template '
+                        f'form by user {user}: {e}'
+                    )
+                    return
+                if not user.is_superuser and not user.is_manager(organization):
+                    logger.warn(
+                        'Detected tampering in clone template '
+                        f'form by user {user}: not authorized '
+                        f'to operate on {validated_org}.'
+                    )
+                    return
+            elif organization == '':
+                validated_org = None
+                if not user.is_superuser:
+                    logger.warn(
+                        'Detected tampering in clone template '
+                        f'form by user {user}: not authorized to '
+                        f'clone a template and set it to shared.'
+                    )
+                    return
+
+            for template in queryset:
+                try:
+                    clone = template.clone(user)
+                    # user has access to multiple orgs
+                    if organization is not None:
+                        clone.organization = validated_org
+                    # user has access only to one org
+                    else:
+                        clone.organization = template.organization
+                    clone.save()
+                    create_log_entry(user, clone)
+                except ValidationError as e:
+                    errors.append([template, e])
+
+            if errors:
+                msg = ''
+                for error in errors:
+                    info = ''
+                    for attr, reasons in dict(error[1]).items():
+                        reasons = ', '.join(reasons)
+                        info = f'{attr}: {reasons}'
+                    msg += f'"{error[0].name}", {info} - '
+                msg = msg[0:-3]  # remove trailing ", "
+                view.message_user(
+                    request,
+                    _('There were errors while cloning the following templates. %s')
+                    % msg,
+                    messages.ERROR,
+                )
```

Comment:
![Screenshot from 2024-02-20 13-49-29](https://github.com/openwisp/openwisp-controller/assets/32094433/611fd5fa-f9c1-48c4-9bc6-d0a2020e625a)

This is how the message appears if there are errors on multiple templates. Can we use HTML formatting when crafting the message to be displayed?